Supporting Cassandra 3.0+

The current cassandra2-cql layer errors when used with new versions of cassandra.
```
$ bin/ycsb run cassandra2-cql -P workloads/workloadc -p hosts=localhost
[WARN]  Running against a source checkout. In order to get our runtime dependencies we'll have to invoke Maven. Depending on the state of your system, this may take ~30-45 seconds
[DEBUG]  Running 'mvn -pl com.yahoo.ycsb:cassandra2-binding -am package -DskipTests dependency:build-classpath -DincludeScope=compile -Dmdep.outputFilterFile=true'
java -cp /Users/m/Desktop/scratch/YCSB/cassandra2/conf:/Users/m/Desktop/scratch/YCSB/cassandra2/target/cassandra2-binding-0.7.0-SNAPSHOT.jar:/Users/m/.m2/repository/io/netty/netty-handler/4.0.27.Final/netty-handler-4.0.27.Final.jar:/Users/m/.m2/repository/org/hdrhistogram/HdrHistogram/2.1.4/HdrHistogram-2.1.4.jar:/Users/m/.m2/repository/io/netty/netty-transport/4.0.27.Final/netty-transport-4.0.27.Final.jar:/Users/m/.m2/repository/org/codehaus/jackson/jackson-core-asl/1.9.4/jackson-core-asl-1.9.4.jar:/Users/m/.m2/repository/io/netty/netty-buffer/4.0.27.Final/netty-buffer-4.0.27.Final.jar:/Users/m/.m2/repository/io/netty/netty-common/4.0.27.Final/netty-common-4.0.27.Final.jar:/Users/m/.m2/repository/com/datastax/cassandra/cassandra-driver-core/2.1.8/cassandra-driver-core-2.1.8.jar:/Users/m/.m2/repository/org/slf4j/slf4j-api/1.6.4/slf4j-api-1.6.4.jar:/Users/m/.m2/repository/org/codehaus/jackson/jackson-mapper-asl/1.9.4/jackson-mapper-asl-1.9.4.jar:/Users/m/.m2/repository/com/codahale/metrics/metrics-core/3.0.2/metrics-core-3.0.2.jar:/Users/m/Desktop/scratch/YCSB/core/target/core-0.7.0-SNAPSHOT.jar:/Users/m/.m2/repository/io/netty/netty-codec/4.0.27.Final/netty-codec-4.0.27.Final.jar:/Users/m/.m2/repository/com/google/guava/guava/14.0.1/guava-14.0.1.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -P workloads/workloadc -p hosts=localhost -t
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -P workloads/workloadc -p hosts=localhost -t
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9042 (com.datastax.driver.core.exceptions.InvalidQueryException: unconfigured table schema_keyspaces))
	at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:193)
	at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:77)
	at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9042 (com.datastax.driver.core.exceptions.InvalidQueryException: unconfigured table schema_keyspaces))
	at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:227)
	at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:86)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1409)
	at com.datastax.driver.core.Cluster.getMetadata(Cluster.java:382)
	at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:180)
	... 2 more
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9042 (com.datastax.driver.core.exceptions.InvalidQueryException: unconfigured table schema_keyspaces))
	at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:193)
	at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:77)
	at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9042 (com.datastax.driver.core.exceptions.InvalidQueryException: unconfigured table schema_keyspaces))
	at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:227)
	at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:86)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1409)
	at com.datastax.driver.core.Cluster.getMetadata(Cluster.java:382)
	at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:180)
	... 2 more
[OVERALL], RunTime(ms), 725.0
[OVERALL], Throughput(ops/sec), 0.0
```
Updating the version of the cassandra driver to 3.0.0 fixes this. As per http://datastax.github.io/java-driver/#compatibility, compatibility with older versions should be unaffected.

--- a/cassandra2/pom.xml
+++ b/cassandra2/pom.xml
@@ -46,8 +46,9 @@ LICENSE file.
     </dependency>
     <dependency>
       <groupId>org.cassandraunit</groupId>
-      <artifactId>cassandra-unit-shaded</artifactId>
-      <version>2.1.9.2</version>
+      <artifactId>cassandra-unit</artifactId>
+      <version>3.0.0.1</version>
+      <classifier>shaded</classifier>
       <scope>test</scope>
     </dependency>
     <dependency>

--- a/cassandra2/src/test/java/com/yahoo/ycsb/db/CassandraCQLClientTest.java
+++ b/cassandra2/src/test/java/com/yahoo/ycsb/db/CassandraCQLClientTest.java
@@ -41,6 +41,7 @@ import com.yahoo.ycsb.workloads.CoreWorkload;
 import org.cassandraunit.CassandraCQLUnit;
 import org.cassandraunit.dataset.cql.ClassPathCQLDataSet;
 import org.junit.After;
+import org.junit.Assume;
 import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.Test;
@@ -63,11 +64,16 @@ public class CassandraCQLClientTest {
   private Session session;
 
   @ClassRule
-  public static CassandraCQLUnit cassandraUnit =
-      new CassandraCQLUnit(new ClassPathCQLDataSet("ycsb.cql", "ycsb"));
+  public static CassandraCQLUnit cassandraUnit = new CassandraCQLUnit(new ClassPathCQLDataSet("ycsb.cql", "ycsb"));
 
   @Before
-  public void setUpClient() throws Exception {
+  public void setUp() throws Exception {
+    // check that this is Java 8+
+    int javaVersion = Integer.parseInt(System.getProperty("java.version").split("\\.")[1]);
+    Assume.assumeTrue(javaVersion >= 8);
+
+    session = cassandraUnit.getSession();
+
     Properties p = new Properties();
     p.setProperty("hosts", HOST);
     p.setProperty("port", Integer.toString(PORT));
@@ -81,14 +87,11 @@ public class CassandraCQLClientTest {
     client.init();
   }
 
-  @Before
-  public void setSession() {
-    session = cassandraUnit.getSession();
-  }
-
   @After
   public void tearDownClient() throws Exception {
-    client.cleanup();
+    if (client != null) {
+      client.cleanup();
+    }
     client = null;
   }
 
@@ -96,7 +99,9 @@ public class CassandraCQLClientTest {
   public void clearTable() throws Exception {
     // Clear the table so that each test starts fresh.
     final Statement truncate = QueryBuilder.truncate(TABLE);
-    cassandraUnit.getSession().execute(truncate);
+    if (cassandraUnit != null) {
+      cassandraUnit.getSession().execute(truncate);
+    }
   }
 
   @Test

--- a/pom.xml
+++ b/pom.xml
@@ -74,7 +74,7 @@ LICENSE file.
     <accumulo.version>1.6.0</accumulo.version>
     <cassandra.version>1.2.9</cassandra.version>
     <cassandra.cql.version>1.0.3</cassandra.cql.version>
-    <cassandra2.cql.version>2.1.8</cassandra2.cql.version>
+    <cassandra2.cql.version>3.0.0</cassandra2.cql.version>
     <geode.version>1.0.0-incubating.M1</geode.version>
     <infinispan.version>7.2.2.Final</infinispan.version>
     <kudu.version>0.6.0</kudu.version>

